aten/src/ATen/cuda/CUDAGraph.h

```diff
@@ -24,15 +26,19 @@ struct TORCH_CUDA_CPP_API CUDAGraph {
   static void inc_pending_event_queries();
   static void dec_pending_event_queries();
   static int num_pending_event_queries();
-  void capture_begin(MempoolId_t pool={0, 0}, cudaStreamCaptureMode capture_mode = cudaStreamCaptureModeGlobal);
+  void register_generator_state(c10::intrusive_ptr<at::CUDAGeneratorState> state);
```

Comment:
It's pretty interesting that we're taking ownership of the intrusive ptr here. Not sure if this could cause memory problems, I guess not.